.github/workflows/build-release.yml

```
@@ -0,0 +1,40 @@
+name: Build or Release Image
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+**'
+    branches:
+      - 'mayastor'
+
+jobs:
+  plugin:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+        id: go
+
+      - name: Set Image Tag
+        run: |
+          if [[ "${GITHUB_REF}" == refs/tags/* ]]; then
+            TAG="${GITHUB_REF#refs/tags/}"
```

nit: could use the github var

```suggestion
          if [[ "${{ github.event_name }}" = "tag" ]]; then
            TAG="${{ github.ref_name }}"
```